src/modules/file-restoration.ts

```
@@ -0,0 +1,413 @@
+      };
+    } catch (error) {
+      return { exists: true };
+    }
+  }
+
+  /**
+   * Cross-platform file path validation
+   */
+  static validateFilePath(filePath: string): { isValid: boolean; error?: string } {
+    // Check for invalid characters (common across platforms)
+    const invalidChars = /[<>:"/\\|?*\x00-\x1f]/;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)